File: src/example/nodes/topic_publisher.py
#!/usr/bin/env python

"""This script posts a contrived ExampleMessage to a topic"""

import random

import rospy
from example.msg import ExampleMessage

NODE_NAME = 'topic_publisher'
TOPIC_NAME = 'topic'


def run(first_name, last_name):
    rospy.init_node(NODE_NAME)

    pub = rospy.Publisher(TOPIC_NAME, ExampleMessage, queue_size=10)

    rate = rospy.Rate(10)

    while not rospy.is_shutdown():
        data = ExampleMessage(
            first_name=first_name,
            last_name=last_name,
            age=random.randrange(1, 100),
            score=random.choice([420, 1337, 8008135])
        )

        rospy.loginfo(data)

        pub.publish(data)

        rate.sleep()


if __name__ == '__main__':
    try:
        run('Edward', 'Beech')
    except rospy.ROSInterruptException:
        pass

File: src/example/nodes/topic_publisher.cpp
#include "ros/ros.h"
#include "example/ExampleMessage.h"

#include "HelloWorld/HelloWorldPublisher.h"


using namespace eprosima::fastrtps;
using namespace rtps;


const std::string NODE_NAME = "topic_publisher";
const std::string TOPIC_NAME = "topic";

unsigned int rand_between(int min, int max) {
    return rand() % (max - min);
}

int main(int argc, char **argv) {
    HelloWorldPublisher helloWorldPublisher;

    helloWorldPublisher.init();

    ros::init(argc, argv, NODE_NAME);

    ros::NodeHandle nh;

    ros::Publisher pub = nh.advertise<example::ExampleMessage>(TOPIC_NAME, 1000);

    ros::Rate loop_rate(10);

    while (ros::ok()) {
        example::ExampleMessage msg;

        int scores[3] = {420, 1337, 8008135};

        msg.first_name = "Edward";
        msg.last_name = "Beech";
        msg.age = rand_between(1, 100);
        msg.score = scores[rand_between(0, 2)];

        ROS_INFO(
                "first_name=%s, last_name=%s, age=%i, score=%i",
                msg.first_name.c_str(),
                msg.last_name.c_str(),
                msg.age,
                msg.score
        );

        pub.publish(msg);

        ros::spinOnce();

        loop_rate.sleep();
    }

    return 0;
}